### PR TITLE
Use CSS style for colors

### PR DESCRIPTION
**Pull request by patrislav1**
This PR is trying to use CSS styles for recurring style attributes (primarily for colors) to 1) reduce SVG size and make it more readable and 2) make colors easily adjustable in the SVG code for all characters that use it. This is WIP, I thought I could get away with using a single set of color styles for both background and foreground, but turns out that setting the stroke color affect the text rendering.

### CODE REVIEW

**Comment by wader** [comment]
Nice, good idea. When messing around with these thing you really appreciate having good tests :) so please add more if you like

**Comment by wader** [comment]
Some possible things to look into:
- `<g>` can be used to inherit attributes and transformations
- `<use>` be be used to duplicate a  `<g>`

But as you also mentioned i think we should probably stay away from too crazy tricks to keep the svg easily editable. For  
example i wonder if the background could be a small embedded png that is stretched? :)

File: svgscreen/template.svg [hunk 1]
```diff
@@ -28,16 +28,51 @@ whitespace only tspan:s preventing underline to be properly rendered.
...
+{{- if anyTrue $.Background.ANSIUsed}}
+        <!-- Background ANSI colors -->
+{{- end}}
+{{- range $k, $v := $.ANSIColors -}}
+        {{- if index $.Background.ANSIUsed $k}}
+        .ba{{$k}} { stroke: {{$v}}; fill: {{$v}}; }
+        {{- end}}
+{{- end}}
+{{- if anyTrue $.Foreground.ANSIUsed}}
+        <!-- Foreground ANSI colors -->
```
Comment (wader) [inline svgscreen/template.svg:45]: Intension it that comment should end up in the output? otherwise can use `{{- /* ... */ -}}`

File: svgscreen/svgscreen.go [hunk 2]
```diff
@@ -235,12 +257,33 @@ func (s *Screen) setupBgRects() {
...
+		result[v] = k
+	}
+	*clsTable = result
+}
+
 func (s *Screen) Render(w io.Writer) error {
 	t := template.New("")
 	t.Funcs(template.FuncMap{
 		"base64": func(bs []byte) string { return base64.RawStdEncoding.EncodeToString(bs) },
+		"anyTrue": func(arr [16]bool) bool {
```
Comment (wader) [inline svgscreen/svgscreen.go:272]: Could this be `arr []bool`? btw i usually use the convention that slice variables names end with `s`, ex `bs` for byte slice

... now i see `ANSIUsed`  is `[16]bool`. If you want it to be a slice you have to assign `ANSIUsed` using `make([]bool, 16)` etc.

Maybe what mostly confused me was the name. Keep type and more explicit name like `anyColorTrue` etc?

**Comment by wader** [comment]
Output looks good, can merge if you want 👍 had some minor code comments but no biggies.

For new features, regressions or some tricky edge cases i usually try to add some kind of test

File: svgscreen/template.svg (hunk 1 above)
Comment (patrislav1) [reply svgscreen/template.svg:45]: yep, it's intentional, to be clearer what the classes are for. but maybe it's not necessary, WDYT?

File: svgscreen/svgscreen.go (hunk 2 above)
Comment (patrislav1) [reply svgscreen/svgscreen.go:272]: to keep it simple I'll just change the name

File: svgscreen/template.svg (hunk 1 above)
Comment (wader) [reply svgscreen/template.svg:45]: let's keep it!

File: svgscreen/svgscreen.go (hunk 2 above)
Comment (wader) [reply svgscreen/svgscreen.go:272]: 👍

**Comment by patrislav1** [comment]
:rocket:
